examples/simple-cluster.yaml

```diff
@@ -1,38 +1,108 @@
+metadata:
+  name: my-cluster
+  namespace: default
+---
+apiVersion: cluster.x-k8s.io/v1alpha2
 kind: Cluster
 metadata:
   name: my-cluster
+  namespace: default
 spec:
-  clusterNetwork:
-    services:
-      cidrBlocks: ["10.96.0.0/12"]
-    pods:
-      cidrBlocks: ["192.168.0.0/16"]
-    serviceDomain: "cluster.local"
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+    kind: DockerCluster
+    name: my-cluster
+    namespace: default
 ---
-apiVersion: "cluster.k8s.io/v1alpha1"
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: DockerMachine
+metadata:
+  name: controlplane-0
+  namespace: default
+---
+apiVersion: cluster.x-k8s.io/v1alpha2
 kind: Machine
 metadata:
-  name: my-control-plane
   labels:
-    cluster.k8s.io/cluster-name: my-cluster
-    set: "controlplane"
+    cluster.x-k8s.io/cluster-name: my-cluster
+    cluster.x-k8s.io/control-plane: "true"
+  name: controlplane-0
+  namespace: default
 spec:
-  providerSpec: {}
-  versions:
-    controlPlane: "v1.14.2"
-    kubelet: "v1.14.2"
+  bootstrap:
+    configRef:
+      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+      kind: KubeadmConfig
+      name: controlplane-0-config
+      namespace: default
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+    kind: DockerMachine
+    name: controlplane-0
+    namespace: default
 ---
-apiVersion: "cluster.k8s.io/v1alpha1"
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+kind: KubeadmConfig
+metadata:
+  name: controlplane-0-config
+  namespace: default
+spec:
+  clusterConfiguration:
+    apiServer: {}
+    certificatesDir: ""
+    controlPlaneEndpoint: ""
+    controllerManager: {}
+    dns:
+      type: ""
+    etcd: {}
+    imageRepository: ""
+    kubernetesVersion: ""
+    networking:
+      dnsDomain: ""
+      podSubnet: ""
+      serviceSubnet: ""
+    scheduler: {}
+  initConfiguration:
+    localAPIEndpoint:
+      advertiseAddress: ""
+      bindPort: 0
+    nodeRegistration: {}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: DockerMachine
+metadata:
+  name: worker-0
+  namespace: default
+---
+apiVersion: cluster.x-k8s.io/v1alpha2
 kind: Machine
 metadata:
-  name: worker
   labels:
-    cluster.k8s.io/cluster-name: my-cluster
-    set: "worker"
-  annotations:
-    name: "my-cluster-worker"
+    cluster.x-k8s.io/cluster-name: my-cluster
+  name: worker-0
+  namespace: default
+spec:
+  bootstrap:
+    configRef:
+      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+      kind: KubeadmConfig
+      name: worker-0-config
+      namespace: default
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+    kind: DockerMachine
+    name: worker-0
+    namespace: default
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+kind: KubeadmConfig
+metadata:
+  name: worker-0-config
+  namespace: default
 spec:
-  providerSpec: {}
-  versions:
-    kubelet: "v1.14.2"
+  joinConfiguration:
+    discovery: {}
```

Comment:
with the above change you can remove `discovery` and `nodeRegistration`

Reply:
Seems like nodeRegistration is required by the validating-webhook
```
error: error validating "mvp/workload/cluster-test/simple-cluster.yaml": error validating data: ValidationError(KubeadmConfig.spec.joinConfiguration): missing required field "nodeRegistration" in io.x-k8s.cluster.bootstrap.v1alpha2.KubeadmConfig.spec.joinConfiguration; if you choose to ignore these errors, turn validation off with --validate=false
```

Reply:
Seems like nodeRegistration is required by the validating-webhook
```
error: error validating "cluster-test/simple-cluster.yaml": error validating data: ValidationError(KubeadmConfig.spec.joinConfiguration): missing required field "nodeRegistration" in io.x-k8s.cluster.bootstrap.v1alpha2.KubeadmConfig.spec.joinConfiguration; if you choose to ignore these errors, turn validation off with --validate=false
```